src/Language/Haskell/Liquid/Bare/OfType.hs

```diff
@@ -148,6 +143,24 @@ ofBRType appRTAlias resolveReft
     go_syms
       = secondM ofBSort
 
+    goRFun bounds _ (RApp c ps _ _) t | Just bnd <- M.lookup c bounds 
+      = makeBound bnd (bar <$> ps) <$> go t
+    goRFun _ x t1 t2
+      = rFun x <$> go t1 <*> go t2
+
+    bar (RVar x _) = x
+    bar _          = errorstar "BAR"   
```

Comment:
`errorstar` is just a [wrapper around `error`](https://github.com/ucsd-progsys/liquid-fixpoint/blob/b1592e1d8189a8f8ba5c2920e0c8eaffdc9e607d/src/Language/Fixpoint/Misc.hs#L88).

`ofBRType` is already in `BareM`, so we should throw a proper `Error` via `throwError` instead of using `errorstar` here.